### DISCUSSION TITLE
Help me to understand the app

### DISCUSSION THREAD

**Discussion by mavo** [reinhard-brandstaedter/solarflow-control#133] [Q&A]
Hello @reinhard-brandstaedter,

I have to say sorry for 2 things:
1) because I think I must be the most annoying guy in the internet for you lately
2) because I did open issues and never saw the discussion section which most of my texts would have been better placed.

So - SORRY - I hope you still will answer, as I really like this app and I am using it for a couple of days now where it mostly is what I need and dreamed of without the need to code myself.

I have another question on how the system behaves 🙈 
So, lets say we have the situation (as I had it today) with the following values:
- Demand from home: 500W
- Direct solar input to inverter: 100W
- SF Hub Solar input: 400W
- Min Charge configured: 200W

In this setup, the app did something I did not expect to happen: Whenever the Battery had 1% more then its lowest allowed state (10%) it started to feed the house... Isn't this supposed to not happen? 
As far as I understood how it works is, that it will charge the battery with its 200W and allocate all the remainder to the house. But this is not what happend, and it was kinda confusing to me.

Why this is relevant? As my battery seems to not be calibrated 100% as of yet, it "jumped" from 10% SoC to around 40% with only putting in ~0,3kWh (which is less then 10% of my actual capacity) and as it then started to discharge, it all of the sudden jumped to 0% which I thought should not happen but is maybe connected to not 100% calibrated. Hoewever, as I cannot trust the %SoC yet, I do not want to have the battery deeply discharged by accident.

So my question is: Is this something I missunderstood and it actually should happen the way it did? Or is this down to my config of allowing discharge of the battery in the middle of the day? (I know, its kinda contradicting to have min charge of 200W and a wish to discharge when sun is still available, but the min_charge was set because I saw the battery jumping and discharging whenever it had 1% more than its min SoC which I wanted to prevent by setting the min_charge with the described outcome above).

Regards and thanks in advance for the time you invest to answer me again.
MaVo

**Comment by reinhard-brandstaedter** [comment]
Hi @mavo ,

no need to be sorry for anything, questions, feedback and discussions is what makes things better!

In your described situation, the behavior should be as you described. In short: the control script should try to get ~400W from the hub (500W - 100W). Then depending on the state of the hub:

1. ) the battery had not been full for you specified interval: not give anything to home, use everything to charge (this is likely not the situation you ran into)

2.) the battery had recently been full and is allowed to discharge (likely your case): the hub should either feed the smaller value of 400W and your MIN_DISCHARGE_POWER (in case you have no sunrise/sunset offset, or solarinput_hub - MIN_CHARGE => 200W to home

What might cause your observation is that you've set a min SoC Level of 10%. I think we are all running with a minSoC of 0% and solarflow-control is actually not honoring the minSoC ( I should mention that somewhere, sorry).
SF control "maintains" the calibration by going 100% full to completely empty and it will switch behavior on that also.

E.g.: once the battery is at 100% it goes into a target mode "discharging" (you can see that in MQTT topic (solarflow-hub/<id>/control/batteryTarget = "discharging"). This mode means it will discharge (with the above parameters) until the battery reaches 0%. It can still charge throughout the day (and can reach 100% again) but it will try to get to zero even if the battery hasn't reached 100% for longer than the specified charge-through interval.
If the target mode is in "charging" mode then the hub will only be discharge (again according to params) if the battery had been at 100% within the charge-through interval.

I know that sounds a bit confusing, but that way we can maintain calibration (full 0-100-0 cycles), and also work with situations where the battery gets from >0% to 100% again throughout the day. At some rainy days it will still get down to 0. 

If you post your log output you will find indicators like this:
```
Based on time, solarpower ( 0.0W) minimum charge power (125W) and bypass state (False), hub could contribute 150.0W - Decision path: 2.1.
```
The decision path helps identifying what exactly led to the actions.

Keep posting questions & ideas!
